htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Table.java

```diff
@@ -142,7 +142,10 @@ public WebElement getCellAt(int i, int j) {
      * Returns list of maps where keys are table headings and values are table row elements ({@code <td>}).
      */
     public List<Map<String, WebElement>> getRowsMappedToHeadings() {
-        return getRowsMappedToHeadings(getHeadingsAsString());
+        return getRows().stream()
+                .map(row -> row.stream()
+                        .collect(toMap(e -> getHeadingsAsString().get(row.indexOf(e)), identity())))
```

Comment:
Would it be best to cache the HeadingsAsString list - otherwise for each row you will have to go back to the browser to get the elements and their text each time?